src/main/java/org/checkerframework/specimin/Slicer.java

```diff
@@ -0,0 +1,212 @@
+package org.checkerframework.specimin;
+
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.NodeList;
+import com.github.javaparser.ast.body.TypeDeclaration;
+import com.github.javaparser.ast.stmt.BlockStmt;
+import com.github.javaparser.resolution.Resolvable;
+import com.github.javaparser.resolution.UnsolvedSymbolException;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.IdentityHashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.checkerframework.specimin.unsolved.AddInformationResult;
+import org.checkerframework.specimin.unsolved.UnsolvedSymbolAlternates;
+import org.checkerframework.specimin.unsolved.UnsolvedSymbolGenerator;
+
+public class Slicer {
```

Comment:
I would expect a slicer to take a program and a dependency map as input and output the slice. It's not clear that this class actually has that contract, though: the constructor has some other arguments, and the slice() method is void. How does a user of this class get the output? The Javadoc should explain how the class is intended to be used.